### PR TITLE
feat(autoware_planning_msgs): create PoseWithUuid and add uuid to VectorMapRoute 

### PR DESCRIPTION
**Pull request by kosuke55**
Signed-off-by: kosuke55 <kosuke.tnp@gmail.com>

## Description

see https://github.com/autowarefoundation/autoware/discussions/2983 for details

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].
- [x] The PR has been properly tested.
- [x] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by yukkysaito** [comment]
what reason did you change from uuid to string? :pray: 
https://github.com/autowarefoundation/autoware_msgs/pull/42/commits/590781eef55d8995f9fcce89e42aa7fd0390676a

**Comment by xmfcx** [comment]
@kosuke55 could you rename `VectorMap` to `Lanelet` since now https://github.com/autowarefoundation/autoware_msgs/pull/41 is merged?

Sorry for the extra work.

File: autoware_planning_msgs/msg/PoseWithUUID.msg [hunk 1]
```diff
@@ -0,0 +1,2 @@
+geometry_msgs/Pose pose
+string uuid
```
Comment (kosuke55) [inline autoware_planning_msgs/msg/PoseWithUUID.msg:2]: Uuid

**Comment by kosuke55** [comment]
@xmfcx 
thank you for your suggestion. After discussion, we preferred unique_identifier_msgs/UUID msg.

@isamu-takagi @yukkysaito @mitsudome-r 
Could you please check this PR?

**Review by isamu-takagi — APPROVED** [review]
LGTM

**Review by xmfcx — APPROVED** [review]

**Comment by xmfcx** [comment]
Oh just realized, why did you prefer https://github.com/ros2/unique_identifier_msgs over https://github.com/ros2/unique_identifier/tree/main/uuid_msgs ?

**Pull request merged**